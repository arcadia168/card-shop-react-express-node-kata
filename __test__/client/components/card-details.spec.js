import React from 'react';
import CardDetails from '../../../src/client/components/card-details';
import { mount } from 'enzyme';

describe('Card Details', () => {
    const render = customProps => {
        const props = {
            ...customProps,
        }
        return mount(<CardDetails {...props} />);
    }
    it('Renders a a basic card details container', () => {
        const component = render();
        expect(
            component.find('[data-testid="card-details__container"]').exists()
        ).toBeTruthy();
    });

    it('Renders a row to house the card details', () => {
        const component = render();
        expect(
            component.find('[data-testid="card-details__row"]').exists()
        ).toBeTruthy();
    });

    it('Renders two columns to house the card details', () => {
        const component = render();
        expect(
            component.find('[data-testid="card-details__column-1"]').exists()
        ).toBeTruthy();

        expect(
            component.find('[data-testid="card-details__column-2"]').exists()
        ).toBeTruthy();
    });
});
