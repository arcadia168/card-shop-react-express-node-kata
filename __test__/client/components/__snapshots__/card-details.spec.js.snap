// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Details renders the app as expected 1`] = `
<div
  className="card-details__container container"
  data-testid="card-details__container"
>
  <div
    className="card-details__row row"
    data-testid="card-details__row"
  >
    <div
      className="card-details__column-1 col"
      data-testid="card-details__column-1"
    >
      <img
        className="card-details__column-1 img-fluid"
        data-testid="card-details__image"
        src={null}
      />
    </div>
    <div
      className="card-details__column-1 col"
      data-testid="card-details__column-2"
    >
      <div
        className="card-details__column-2-row-1 row"
        data-testid="card-details__column-2-row1"
      >
        <h1
          className="card-details__title"
          data-testid="card-details__title"
        />
      </div>
    </div>
  </div>
</div>
`;
